scripts/perf-test.sh

```diff
@@ -0,0 +1,191 @@
+  producerPerf broker1:9092 ${TOPIC} ${NUM_RECORDS}
+  doDeleteTopic broker1:9092 perf-test
+}
+
+
+doPerfKroxyNoFilters () {
+  local TOPIC
+  local CFG
+  TOPIC=perf-test
+  CFG=no-filters.yaml
+  echo -e "${GREEN}Running Kroxylicious No Filter ${NOCOLOR}"
+
+  KROXY_CONFIG=${CFG} runDockerCompose up --detach --wait kroxy
+
+  doCreateTopic kroxy:9092 ${TOPIC}
+
+  echo -e "${YELLOW}Running warm up${NOCOLOR}"
+  warmUp kroxy:9092 ${TOPIC} > /dev/null
+  echo -e "${YELLOW}Running test${NOCOLOR}"
+  producerPerf kroxy:9092 ${TOPIC} ${NUM_RECORDS}
+
+  doDeleteTopic kroxy:9092 ${TOPIC}
+  KROXY_CONFIG=${CFG} runDockerCompose rm -s -f kroxy
+}
+
+doPerfKroxyTransformFilter () {
+  local TOPIC
+  local CFG
+  TOPIC=perf-test
+  CFG=transform-filter.yaml
+
+  echo -e "${GREEN}Running Kroxylicious Transform Filter ${NOCOLOR}"
+
+  KROXY_CONFIG=${CFG} runDockerCompose up --detach --wait kroxy
+  doCreateTopic kroxy:9092 ${TOPIC}
+
+  echo -e "${YELLOW}Running warm up${NOCOLOR}"
+  warmUp kroxy:9092 ${TOPIC} > /dev/null
+  echo -e "${YELLOW}Running test${NOCOLOR}"
+  producerPerf kroxy:9092 ${TOPIC} ${NUM_RECORDS}
+
+  doDeleteTopic kroxy:9092 ${TOPIC}
+  KROXY_CONFIG=${CFG} runDockerCompose rm -s -f kroxy
+}
+
+doPerfKroxyEnvelopeEncryptionFilter () {
+  local CFG
+  local TOPIC
+  local ENCRYPT
+  ENCRYPT=$1
+  TOPIC=perf-test
+  CFG=envelope-encryption-filter.yaml
+
+  echo -e "${GREEN}Running Kroxylicious Envelope Encryption Filter (encrypted topic ${ENCRYPT}) ${NOCOLOR}"
+
+  KROXY_CONFIG=${CFG} runDockerCompose up --detach --wait kroxy vault
+
+  docker exec vault vault secrets enable transit
+  if [[ ${ENCRYPT} = true ]]; then
+    docker exec vault vault write -f transit/keys/KEK_${TOPIC}
+  fi
+  doCreateTopic kroxy:9092 ${TOPIC}
+
+  echo -e "${YELLOW}Running warm up${NOCOLOR}"
+  warmUp kroxy:9092 ${TOPIC} > /dev/null
+  echo -e "${YELLOW}Running test${NOCOLOR}"
+  producerPerf kroxy:9092 ${TOPIC} ${NUM_RECORDS}
+
+  doDeleteTopic kroxy:9092 ${TOPIC}
+  KROXY_CONFIG=${CFG} runDockerCompose rm -s -f kroxy vault
+}
+
+onExit() {
+  for cmd in "${ON_SHUTDOWN[@]}"
+  do
+    eval ${cmd}
+  done
+}
+
+KAFKA_VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:3.4.0:evaluate -Dexpression=kafka.version -q -DforceStdout)
+STRIMZI_VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:3.4.0:evaluate -Dexpression=strimzi.version -q -DforceStdout)
+export KAFKA_VERSION STRIMZI_VERSION
+
+trap onExit EXIT
+
+echo -e "${YELLOW}Kafka version is ${KAFKA_VERSION}, Strimzi version ${STRIMZI_VERSION}${NOCOLOR}"
+
+# Bring up Kafka
+ON_SHUTDOWN+=("KROXY_CONFIG=unused.yaml runDockerCompose down")
+KROXY_CONFIG=unused.yaml runDockerCompose up --detach --wait kafka
+
+echo -e "${GREEN}Running test cases, number of records = ${NUM_RECORDS}, record size ${RECORD_SIZE}${NOCOLOR}"
+
+doPerfKafkaDirect
+doPerfKroxyNoFilters
+doPerfKroxyTransformFilter
+doPerfKroxyEnvelopeEncryptionFilter true
+doPerfKroxyEnvelopeEncryptionFilter false
+
```

Comment:
Planning to put the results to files, and then do something to put the results on a chart (throughput/latencies) so we can make easy comparisons.